plugins/UsersManager/API.php

```diff
@@ -323,19 +323,27 @@ public function getUsersPlusRole($idSite, $limit = null, $offset = 0, $filter_se
                 $loginsToLimit = $this->model->getUsersWithAccessToSites($adminIdSites);
             }
 
-            list($users, $totalResults) = $this->model->getUsersWithRole($idSite, $limit, $offset, $filter_search, $filter_access, $loginsToLimit);
-
-            foreach ($users as &$user) {
-                $user['superuser_access'] = $user['superuser_access'] == 1;
-                if ($user['superuser_access']) {
-                    $user['role'] = 'superuser';
-                    $user['capabilities'] = [];
-                } else {
-                    list($user['role'], $user['capabilities']) = $this->getRoleAndCapabilitiesFromAccess($user['access']);
-                    $user['role'] = empty($user['role']) ? 'noaccess' : reset($user['role']);
-                }
+            if ($loginsToLimit !== null && empty($loginsToLimit)) {
+                // if the current user is not the superuser, and getUsersWithAccessToSites() returned an empty result,
+                // access is managed by another plugin, and the current user cannot manage any user with UsersManager
+                Common::sendHeader('X-Matomo-Total-Results: 0');
```

Comment:
@MichaelHeerklotz is the header maybe still there from debugging?

Reply:
It is intentional, some lines above there is another "fast" return case that sets the header, and at the bottom the header is set also for the default case. So i adjusted the new return case to work like the others. I did not work with these custom Matomo before however, so I not 100% sure when exactly to set them and how.

Reply:
Cheers, didn't know :)